flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecRank.java

```diff
@@ -250,17 +257,31 @@ protected Transformation<RowData> translateToPlanInternal(
                                 generateUpdateBefore,
                                 outputRankNumber);
             } else if (RankUtil.isTop1(rankRange)) {
-                processFunction =
-                        new FastTop1Function(
-                                ttlConfig,
-                                inputRowTypeInfo,
-                                sortKeyComparator,
-                                sortKeySelector,
-                                rankType,
-                                rankRange,
-                                generateUpdateBefore,
-                                outputRankNumber,
-                                cacheSize);
+                if (enableAsyncState) {
```

Comment:
NIT: I see the logic 
```
if (RankUtil.isTop1(rankRange)) {
 if (enableAsyncState) {
else
```

Repeated later in the file and in the tests. Can we remove the duplicate code, maybe as a static method in the Helper or maybe a provider / factory class.